cmd/list.go

```diff
@@ -45,14 +49,44 @@ func RunList() *cobra.Command {
 				os.Exit(1)
 			}
 
+			if len(torrents) < 1 {
+				fmt.Println("No torrents found.")
+				return
+			}
+
 			fmt.Println(torrents)
-		} else {
+			return
+		}
+
+		// If no torrent filter provided, list all torrents.
+		if len(args) < 1 {
 			torrents, err := qb.GetTorrents()
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "ERROR: could not get torrents %v\n", err)
 				os.Exit(1)
 			}
 
+			if len(torrents) < 1 {
+				fmt.Println("No torrents found.")
+				return
+			}
+
+			printList(torrents)
+		} else if len(args) >= 1 {
+			var state = args[0]
+			var filter = qbittorrent.TorrentFilter(strings.ToLower(state))
+			torrents, err := qb.GetTorrentsFilter(filter)
+
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "ERROR: could not get torrents %v\n", err)
+				os.Exit(1)
+			}
+
+			if len(torrents) < 1 {
+				fmt.Printf("No torrents found with state \"%s\"\n", state)
+				return
+			}
```

Comment:
With that new flag this could then be simplified a bit, and I would inverse the logic. So `if state != ""` then do the filtering, else just do `GetTorrents()`.

You can also make it simpler and do the err check outside of the if else, and check torrent len there as well. And making it `if len(torrents) == 0 {` would improve the readability a bit imo.